### PR TITLE
feat(connector): [Paypal] implement vaulting for paypal wallet and cards while purchasing

### PR DESCRIPTION
**Pull request by KiranKBR**
## Type of Change

- [x] New feature
## Description

This PR implements Vaulting while purchasing(Mandate flows) for Paypal connector for Card, Paypal redirection and Paypal SDK payment methods.

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

1. Create a Paypal Card Request (CIT)
```
curl --location 'http://localhost:8080/payments' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: {}' \
--data-raw '{
    "amount": 6540,
    "currency": "USD",
    "confirm": true,
    "capture_method": "automatic",
    "capture_on": "2022-09-10T10:11:12Z",
    "amount_to_capture": 6540,
    "customer_id": "StripeCustomer",
    "email": "guest@example.com",
    "name": "John Doe",
    "phone": "999999999",
    "phone_country_code": "+1",
    "description": "Its my first payment request",
    "authentication_type": "no_three_ds",
    "return_url": "https://duck.com",
    "setup_future_usage": "off_session",
    "customer_acceptance": {
        "acceptance_type": "offline",
        "accepted_at": "1963-05-03T04:07:52.723Z",
        "online": {
            "ip_address": "13.232.74.226",
            "user_agent": "amet irure esse"
        }
    },
    "payment_method": "card",
    "payment_method_type": "credit",
    "payment_method_data": {
        "card": {
            "card_number": "4012000033330026",
            "card_exp_month": "10",
            "card_exp_year": "25",
            "card_holder_name": "joseph Doe",
            "card_cvc": "123"
        }
    },
    "billing": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "state": "California",
            "zip": "94122",
            "country": "US",
            "first_name": "PiX"
        }
    },
    "shipping": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "state": "California",
            "zip": "94122",
            "country": "US",
            "first_name": "PiX"
        }
    },
    "statement_descriptor_name": "joseph",
    "statement_descriptor_suffix": "JS",
    "metadata": {
        "udf1": "value1",
        "new_customer": "true",
        "login_date": "2019-09-10T10:11:12Z"
    }
}'
```
Response
```
{
    "payment_id": "pay_01ATaA81YclH6LewH4i6",
    "merchant_id": "merchant_1730195598",
    "status": "succeeded",
    "amount": 6540,
    "net_amount": 6540,
    "shipping_cost": null,
    "amount_capturable": 0,
    "amount_received": 6540,
    "connector": "paypal",
    "client_secret": "pay_01ATaA81YclH6LewH4i6_secret_vX7C6F7gK7C95VgdECLi",
    "created": "2024-11-03T15:59:28.560Z",
    "currency": "USD",
    "customer_id": "StripeCustomer",
    "customer": {
        "id": "StripeCustomer",
        "name": "John Doe",
        "email": "guest@example.com",
        "phone": "999999999",
        "phone_country_code": "+1"
    },
    "description": "Its my first payment request",
    "refunds": null,
    "disputes": null,
    "mandate_id": null,
    "mandate_data": null,
    "setup_future_usage": "off_session",
    "off_session": null,
    "capture_on": null,
    "capture_method": "automatic",
    "payment_method": "card",
    "payment_method_data": {
        "card": {
            "last4": "0026",
            "card_type": null,
            "card_network": null,
            "card_issuer": null,
            "card_issuing_country": null,
            "card_isin": "401200",
            "card_extended_bin": null,
            "card_exp_month": "10",
            "card_exp_year": "25",
            "card_holder_name": null,
            "payment_checks": null,
            "authentication_data": null
        },
        "billing": null
    },
    "payment_token": null,
    "shipping": {
        "address": {
            "city": "San Fransico",
            "country": "US",
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "zip": "94122",
            "state": "California",
            "first_name": "PiX",
            "last_name": null
        },
        "phone": null,
        "email": null
    },
    "billing": {
        "address": {
            "city": "San Fransico",
            "country": "US",
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "zip": "94122",
            "state": "California",
            "first_name": "PiX",
            "last_name": null
        },
        "phone": null,
        "email": null
    },
    "order_details": null,
    "email": "guest@example.com",
    "name": "John Doe",
    "phone": "999999999",
    "return_url": "https://duck.com/",
    "authentication_type": "no_three_ds",
    "statement_descriptor_name": "joseph",
    "statement_descriptor_suffix": "JS",
    "next_action": null,
    "cancellation_reason": null,
    "error_code": null,
    "error_message": null,
    "unified_code": null,
    "unified_message": null,
    "payment_experience": null,
    "payment_method_type": "credit",
    "connector_label": null,
    "business_country": null,
    "business_label": "default",
    "business_sub_label": null,
    "allowed_payment_method_types": null,
    "ephemeral_key": {
        "customer_id": "StripeCustomer",
        "created_at": 1730649568,
        "expires": 1730653168,
        "secret": "epk_4ee06de7056e47fdae8101ed372be814"
    },
    "manual_retry_allowed": false,
    "connector_transaction_id": "6MP9513504803263N",
    "frm_message": null,
    "metadata": {
        "udf1": "value1",
        "login_date": "2019-09-10T10:11:12Z",
        "new_customer": "true"
    },
    "connector_metadata": null,
    "feature_metadata": null,
    "reference_id": "pay_01ATaA81YclH6LewH4i6_1",
    "payment_link": null,
    "profile_id": "pro_tGKR3r2IFzb9PRFzuLAj",
    "surcharge_details": null,
    "attempt_count": 1,
    "merchant_decision": null,
    "merchant_connector_id": "mca_kqW3BOGnobT4zfviyywR",
    "incremental_authorization_allowed": null,
    "authorization_count": null,
    "incremental_authorizations": null,
    "external_authentication_details": null,
    "external_3ds_authentication_attempted": false,
    "expires_on": "2024-11-03T16:14:28.560Z",
    "fingerprint": null,
    "browser_info": null,
    "payment_method_id": "pm_SdAe0ppJf2EPfKKRbEw3",
    "payment_method_status": "active",
    "updated": "2024-11-03T15:59:36.984Z",
    "charges": null,
    "frm_metadata": null,
    "merchant_order_reference_id": null,
    "order_tax_amount": null,
    "connector_mandate_id": "20a564405w922772c"
}
```
Check for `payment_method_id` in response

2. Create a Payment with vault:
```
curl --location 'http://localhost:8080/payments' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: __' \
--data-raw '{
    "amount": 499,
    "currency": "USD",
    "confirm": true,
    "profile_id": "pro_tGKR3r2IFzb9PRFzuLAj",
    "capture_method": "automatic",
    "customer_id": "StripeCustomer",
    "email": "guest@example.com",
    "off_session": true,
    "recurring_details": {
        "type": "payment_method_id",
        "data": "{__}"
    },
    "authentication_type": "no_three_ds"
}'
```
Response
```
{
    "payment_id": "pay_6e8D6lCZPx1EbBjDVJXf",
    "merchant_id": "merchant_1730195598",
    "status": "succeeded",
    "amount": 499,
    "net_amount": 499,
    "shipping_cost": null,
    "amount_capturable": 0,
    "amount_received": 499,
    "connector": "paypal",
    "client_secret": "pay_6e8D6lCZPx1EbBjDVJXf_secret_59QXsN51aTiHer7uz0Zp",
    "created": "2024-11-03T15:59:43.371Z",
    "currency": "USD",
    "customer_id": "StripeCustomer",
    "customer": {
        "id": "StripeCustomer",
        "name": "John Doe",
        "email": "guest@example.com",
        "phone": "999999999",
        "phone_country_code": "+1"
    },
    "description": null,
    "refunds": null,
    "disputes": null,
    "mandate_id": null,
    "mandate_data": null,
    "setup_future_usage": null,
    "off_session": true,
    "capture_on": null,
    "capture_method": "automatic",
    "payment_method": "card",
    "payment_method_data": {
        "card": {
            "last4": "0026",
            "card_type": null,
            "card_network": null,
            "card_issuer": null,
            "card_issuing_country": null,
            "card_isin": "401200",
            "card_extended_bin": null,
            "card_exp_month": "10",
            "card_exp_year": "25",
            "card_holder_name": "joseph Doe",
            "payment_checks": null,
            "authentication_data": null
        },
        "billing": null
    },
    "payment_token": null,
    "shipping": null,
    "billing": null,
    "order_details": null,
    "email": "guest@example.com",
    "name": "John Doe",
    "phone": "999999999",
    "return_url": null,
    "authentication_type": "no_three_ds",
    "statement_descriptor_name": null,
    "statement_descriptor_suffix": null,
    "next_action": null,
    "cancellation_reason": null,
    "error_code": null,
    "error_message": null,
    "unified_code": null,
    "unified_message": null,
    "payment_experience": null,
    "payment_method_type": "credit",
    "connector_label": null,
    "business_country": null,
    "business_label": "default",
    "business_sub_label": null,
    "allowed_payment_method_types": null,
    "ephemeral_key": {
        "customer_id": "StripeCustomer",
        "created_at": 1730649583,
        "expires": 1730653183,
        "secret": "epk_3ff4d701d6af456fb712548809fa7a59"
    },
    "manual_retry_allowed": false,
    "connector_transaction_id": "2TH799216G1813319",
    "frm_message": null,
    "metadata": null,
    "connector_metadata": null,
    "feature_metadata": null,
    "reference_id": "pay_6e8D6lCZPx1EbBjDVJXf_1",
    "payment_link": null,
    "profile_id": "pro_tGKR3r2IFzb9PRFzuLAj",
    "surcharge_details": null,
    "attempt_count": 1,
    "merchant_decision": null,
    "merchant_connector_id": "mca_kqW3BOGnobT4zfviyywR",
    "incremental_authorization_allowed": null,
    "authorization_count": null,
    "incremental_authorizations": null,
    "external_authentication_details": null,
    "external_3ds_authentication_attempted": false,
    "expires_on": "2024-11-03T16:14:43.371Z",
    "fingerprint": null,
    "browser_info": null,
    "payment_method_id": "pm_SdAe0ppJf2EPfKKRbEw3",
    "payment_method_status": "active",
    "updated": "2024-11-03T15:59:48.133Z",
    "charges": null,
    "frm_metadata": null,
    "merchant_order_reference_id": null,
    "order_tax_amount": null,
    "connector_mandate_id": "20a564405w922772c"
}
```

_______________________________________________________________________________

3. Create a Paypal Redirect Payment: (CIT)
```
curl --location 'http://localhost:8080/payments' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: dev_eIrDR05fM0F4gf5fOhzrRYMpLEWfOWH5uVozax981I2THdaCrgmwiow3dGaGxVKP' \
--data-raw '{
    "amount": 654,
    "currency": "EUR",
    "confirm": true,
    "capture_method": "automatic",
    "capture_on": "2022-09-10T10:11:12Z",
    "amount_to_capture": 654,
    "customer_id": "StripeCustomer223",
    "email": "abcdef123@gmail.com",
    "name": "John Doe",
    "phone": "999999999",
    "phone_country_code": "+65",
    "description": "Its my first payment request",
    
    "return_url": "https://duck.com",
    
    "setup_future_usage": "off_session",
    "billing": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "state": "California",
            "zip": "94122",
            "country": "US"
        }
    },
    "browser_info": {
        "user_agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.110 Safari/537.36",
        "accept_header": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8",
        "language": "nl-NL",
        "color_depth": 24,
        "screen_height": 723,
        "screen_width": 1536,
        "time_zone": 0,
        "java_enabled": true,
        "java_script_enabled": true,
        "ip_address": "127.0.0.1"
    },
    "shipping": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "state": "California",
            "zip": "94122",
            "country": "US",
            "first_name": "John",
            "last_name": "Doe"
        }
    },
    "statement_descriptor_name": "joseph",
    "statement_descriptor_suffix": "JS",
    "metadata": {
        "udf1": "value1",
        "new_customer": "true",
        "login_date": "2019-09-10T10:11:12Z"
    },
    "payment_method": "wallet",
    "payment_method_type": "paypal",
    "payment_method_data": {
        "wallet": {
            "paypal_redirect": {}
        }
    },
    "customer_acceptance": {
        "acceptance_type": "offline",
        "accepted_at": "1963-05-03T04:07:52.723Z",
        "online": {
            "ip_address": "13.232.74.226",
            "user_agent": "amet irure esse"
        }
    }
}'
```

response:
```
{
    "payment_id": "pay_42hrpyxeWTS7NS0NXW0j",
    "merchant_id": "merchant_1724323374",
    "status": "requires_customer_action",
    "amount": 654,
    "net_amount": 654,
    "amount_capturable": 654,
    "amount_received": null,
    "connector": "paypal",
    "client_secret": "pay_42hrpyxeWTS7NS0NXW0j_secret_X48UnGBeTRElSVzcOMvI",
    "created": "2024-08-22T11:04:56.338Z",
    "currency": "EUR",
    "customer_id": "StripeCustomer223",
    "customer": {
        "id": "StripeCustomer223",
        "name": "John Doe",
        "email": "abcdef123@gmail.com",
        "phone": "999999999",
        "phone_country_code": "+65"
    },
    "description": "Its my first payment request",
    "refunds": null,
    "disputes": null,
    "mandate_id": null,
    "mandate_data": null,
    "setup_future_usage": "off_session",
    "off_session": null,
    "capture_on": null,
    "capture_method": "automatic",
    "payment_method": "wallet",
    "payment_method_data": {
        "wallet": {},
        "billing": null
    },
    "payment_token": null,
    "shipping": {
        "address": {
            "city": "San Fransico",
            "country": "US",
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "zip": "94122",
            "state": "California",
            "first_name": "John",
            "last_name": "Doe"
        },
        "phone": null,
        "email": null
    },
    "billing": {
        "address": {
            "city": "San Fransico",
            "country": "US",
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "zip": "94122",
            "state": "California",
            "first_name": null,
            "last_name": null
        },
        "phone": null,
        "email": null
    },
    "order_details": null,
    "email": "abcdef123@gmail.com",
    "name": "John Doe",
    "phone": "999999999",
    "return_url": "https://duck.com/",
    "authentication_type": "no_three_ds",
    "statement_descriptor_name": "joseph",
    "statement_descriptor_suffix": "JS",
    "next_action": {
        "type": "redirect_to_url",
        "redirect_to_url": "http://localhost:8080/payments/redirect/pay_42hrpyxeWTS7NS0NXW0j/merchant_1724323374/pay_42hrpyxeWTS7NS0NXW0j_1"
    },
    "cancellation_reason": null,
    "error_code": null,
    "error_message": null,
    "unified_code": null,
    "unified_message": null,
    "payment_experience": null,
    "payment_method_type": "paypal",
    "connector_label": null,
    "business_country": null,
    "business_label": "default",
    "business_sub_label": null,
    "allowed_payment_method_types": null,
    "ephemeral_key": {
        "customer_id": "StripeCustomer223",
        "created_at": 1724324696,
        "expires": 1724328296,
        "secret": "epk_d7d4396e232a4c83bf734b9091cb4dd6"
    },
    "manual_retry_allowed": null,
    "connector_transaction_id": "8NY95914BG6889731",
    "frm_message": null,
    "metadata": {
        "udf1": "value1",
        "login_date": "2019-09-10T10:11:12Z",
        "new_customer": "true"
    },
    "connector_metadata": null,
    "feature_metadata": null,
    "reference_id": null,
    "payment_link": null,
    "profile_id": "pro_SmhBWSeP2rbSxK4jK1jE",
    "surcharge_details": null,
    "attempt_count": 1,
    "merchant_decision": null,
    "merchant_connector_id": "mca_RCqgqX4n5WsNHdbzwHO6",
    "incremental_authorization_allowed": null,
    "authorization_count": null,
    "incremental_authorizations": null,
    "external_authentication_details": null,
    "external_3ds_authentication_attempted": false,
    "expires_on": "2024-08-22T11:19:56.337Z",
    "fingerprint": null,
    "browser_info": {
        "language": "nl-NL",
        "time_zone": 0,
        "ip_address": "127.0.0.1",
        "user_agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.110 Safari/537.36",
        "color_depth": 24,
        "java_enabled": true,
        "screen_width": 1536,
        "accept_header": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8",
        "screen_height": 723,
        "java_script_enabled": true
    },
    "payment_method_id": "pm_M0zZ4dHkloCyaX6H6vV1",
    "payment_method_status": null,
    "updated": "2024-08-22T11:04:58.745Z",
    "charges": null,
    "frm_metadata": null,
    "merchant_order_reference_id": null
}
```
Check for `payment_method_id` in response
Complete the payment using redirect url.

4. Payment with vault: (MIT)
request:
```
curl --location 'http://localhost:8080/payments' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: dev_eIrDR05fM0F4gf5fOhzrRYMpLEWfOWH5uVozax981I2THdaCrgmwiow3dGaGxVKP' \
--data-raw '{
    "amount": 499,
    "currency": "EUR",
    "confirm": true,
    "profile_id": "pro_SmhBWSeP2rbSxK4jK1jE",
    "capture_method": "automatic",
    "customer_id": "StripeCustomer223",
    "email": "guest@example.com",
    "off_session": true,
    "recurring_details": {
        "type": "payment_method_id",
        "data": "{__}"
    },
    "authentication_type": "no_three_ds"
}'
```

response:
```
{
    "payment_id": "pay_1BPTB4CGBhGuUGEEwMOZ",
    "merchant_id": "merchant_1724323374",
    "status": "succeeded",
    "amount": 499,
    "net_amount": 499,
    "amount_capturable": 0,
    "amount_received": 499,
    "connector": "paypal",
    "client_secret": "{_}",
    "created": "2024-08-22T11:06:48.700Z",
    "currency": "EUR",
    "customer_id": "StripeCustomer223",
    "customer": {
        "id": "StripeCustomer223",
        "name": "John Doe",
        "email": "guest@example.com",
        "phone": "999999999",
        "phone_country_code": "+65"
    },
    "description": null,
    "refunds": null,
    "disputes": null,
    "mandate_id": null,
    "mandate_data": null,
    "setup_future_usage": null,
    "off_session": null,
    "capture_on": null,
    "capture_method": "automatic",
    "payment_method": "wallet",
    "payment_method_data": null,
    "payment_token": null,
    "shipping": null,
    "billing": null,
    "order_details": null,
    "email": "guest@example.com",
    "name": "John Doe",
    "phone": "999999999",
    "return_url": null,
    "authentication_type": "no_three_ds",
    "statement_descriptor_name": null,
    "statement_descriptor_suffix": null,
    "next_action": null,
    "cancellation_reason": null,
    "error_code": null,
    "error_message": null,
    "unified_code": null,
    "unified_message": null,
    "payment_experience": null,
    "payment_method_type": "paypal",
    "connector_label": null,
    "business_country": null,
    "business_label": "default",
    "business_sub_label": null,
    "allowed_payment_method_types": null,
    "ephemeral_key": {
        "customer_id": "StripeCustomer223",
        "created_at": 1724324808,
        "expires": 1724328408,
        "secret": "epk_34b80fb13f764e80b2ac0054b35fcec4"
    },
    "manual_retry_allowed": false,
    "connector_transaction_id": "2LR05925G7676561P",
    "frm_message": null,
    "metadata": null,
    "connector_metadata": null,
    "feature_metadata": null,
    "reference_id": null,
    "payment_link": null,
    "profile_id": "pro_SmhBWSeP2rbSxK4jK1jE",
    "surcharge_details": null,
    "attempt_count": 1,
    "merchant_decision": null,
    "merchant_connector_id": "mca_RCqgqX4n5WsNHdbzwHO6",
    "incremental_authorization_allowed": null,
    "authorization_count": null,
    "incremental_authorizations": null,
    "external_authentication_details": null,
    "external_3ds_authentication_attempted": false,
    "expires_on": "2024-08-22T11:21:48.700Z",
    "fingerprint": null,
    "browser_info": null,
    "payment_method_id": "pm_M0zZ4dHkloCyaX6H6vV1",
    "payment_method_status": "active",
    "updated": "2024-08-22T11:06:52.436Z",
    "charges": null,
    "frm_metadata": null,
    "merchant_order_reference_id": null
}
```
- Note: We have to test test same flow for Paypal SDK. 

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: crates/router/src/connector/paypal/transformers.rs [hunk 1]
```diff
@@ -175,18 +176,30 @@ pub struct ShippingName {
...
     name: Secret<String>,
     number: Option<cards::CardNumber>,
     security_code: Option<Secret<String>>,
-    attributes: Option<ThreeDsSetting>,
+    attributes: Option<CardRequestAttributes>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct CardVaultStruct {
+    vault_id: String,
```
Comment (SanchithHegde) [inline crates/router/src/connector/paypal/transformers.rs:190]: We can make this a `Secret` if this is considered sensitive information.

File: crates/router/src/connector/paypal/transformers.rs [hunk 2]
```diff
@@ -230,8 +243,59 @@ pub enum ShippingPreference {
...
-pub struct PaypalRedirectionRequest {
+#[serde(untagged)]
+pub enum PaypalRedirectionRequest {
+    PaypalRedirectionStruct(PaypalRedirectionStruct),
+    PaypalVaultStruct(PaypalVaultStruct),
+}
+#[derive(Debug, Serialize, Deserialize)]
+pub struct PaypalVaultStruct {
+    vault_id: String,
+}
```
Comment (SanchithHegde) [inline crates/router/src/connector/paypal/transformers.rs:254]: How is this different from `CardVaultStruct`?

File: crates/router/src/connector/paypal/transformers.rs [hunk 3]
```diff
@@ -410,26 +484,39 @@ impl TryFrom<&PaypalRouterData<&types::PaymentsAuthorizeRouterData>> for PaypalP
...
-                    attributes,
-                }));
+                let payment_source = Some(PaymentSourceItem::Card(CardRequest::CardRequestStruct(
+                    CardRequestStruct {
+                        billing_address: get_address_info(item.router_data.get_optional_billing())?,
+                        expiry,
+                        name: item
+                            .router_data
+                            .get_optional_billing_full_name()
+                            .unwrap_or(Secret::new("".to_string())),
```
Comment (SanchithHegde) [inline crates/router/src/connector/paypal/transformers.rs:501]: Are we okay with sending empty strings for name?

File: crates/router/src/connector/paypal/transformers.rs [hunk 4]
```diff
@@ -1412,10 +1671,13 @@ impl<F>
...
                 resource_id: types::ResponseId::ConnectorTransactionId(item.response.id),
-                redirection_data: Some(paypal_threeds_link((
+                redirection_data: match paypal_threeds_link((
                     link,
                     item.data.request.complete_authorize_url.clone(),
-                ))?),
+                )) {
+                    Ok(url) => Some(url),
+                    Err(_) => None,
+                },
```
Comment (SanchithHegde) [inline crates/router/src/connector/paypal/transformers.rs:1680]: Are we okay with doing nothing when we receive an error? Can we at least log it?

File: crates/router/src/connector/paypal/transformers.rs [hunk 5]
```diff
@@ -542,10 +663,131 @@ impl TryFrom<&PaypalRouterData<&types::PaymentsAuthorizeRouterData>> for PaypalP
...
             domain::PaymentMethodData::MandatePayment => {
-                Err(errors::ConnectorError::NotImplemented(
-                    utils::get_unimplemented_payment_method_error_message("Paypal"),
-                )
-                .into())
+                let payment_method_type = item
+                    .router_data
+                    .get_recurring_mandate_payment_data()?
+                    .payment_method_type
+                    .ok_or_else(missing_field_err("mandate_id"))?;
```
Comment (swangi-kumari) [inline crates/router/src/connector/paypal/transformers.rs:670]:
```suggestion
                    .ok_or_else(missing_field_err("payment_method_type"))?;
```

**Review by swangi-kumari — DISMISSED** [review]

**Comment by swangi-kumari** [comment]
@KiranKBR 
Please resolve the conflicts.

**Review by swangi-kumari — DISMISSED** [review]

**Review by AkshayaFoiger — DISMISSED** [review]

**Review by SanchithHegde — DISMISSED** [review]

**Review by deepanshu-iiitu — DISMISSED** [review]

**Review by sai-harsha-vardhan — APPROVED** [review]

**Review by deepanshu-iiitu — APPROVED** [review]

**Review by sai-harsha-vardhan — APPROVED** [review]

**Pull request merged**